labelbox/orm/db_object.py

```diff
@@ -244,3 +252,20 @@ def delete(self):
         a call to this you should not use this DB object anymore.
         """
         type(self).bulk_delete([self])
+
+
+
+def beta(fn):
+    def wrapper(self, *args, **kwargs):
+        if not isinstance(self, DbObject):
+            raise TypeError("Cannot decorate functions that are not functions of `DbOjects` with `beta` decorator")
+        if not self.client.enable_beta:
+            logger.warning(
+                f"This function {fn.__name__} relies on a beta feature in the api. This means that the interface could change."
+                " Set `enable_beta=True` in the client to silence this warning.")
+            self.client.endpoint = self.client.endpoint.replace("/graphql", "/_gql")
```

Comment:
I do not like modifying `self.client`

1. on principal seems a little hacky
2. if `fn` throws and exception the clean up code `self.client.endpoint` won't get run and restored


we could fix that in a try/finally of some sort but even then I would prefer

that `beta` functions don't execute and instead utilize their own method `execute_beta` or something like that which has the proper end point. Since `beta` methods are all annotated with a decotator it shouldn't be too hard to go back and change them to regular `execute` if we need to.


Also if they don't have `enable_beta` as true should we just throw?

The current behavior seems more like `warn_on_beta`

Reply:
Maybe cursor pagination can take an execute function then as opposed to being given the client

Reply:
this is all being managed by the beta decorator. Adding here for context.